### PR TITLE
Add check for crashes during E2E tests

### PR DESCRIPTION
**Pull request by robertschaedler3**
## Description

Add a check for crashes during E2E tests and fail E2E job with error messages. Error messages appear as annotations under the workflow summary and contain information about the log file/line number/previous log message where the crash was found in the logs. 

For example:

image

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: .github/actions/check-logs/action.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        import glob
+
+        def error(title, file, line, msg):
+            print(f"::error file={file},line={line},title={title}::{msg}")
+
+        input_glob = "./.github/actions/check-logs/temp/*.log"
+        # input_glob = os.environ.get('logs')
+        log_files = glob.glob(input_glob)
+
+        crashes = ['SIGSEGV', 'SIGFPE', 'SIGILL', 'SIGABRT', 'SIGBUS', 'Interrupt signal']
```
Comment (MariusNi) [inline .github/actions/check-logs/action.yml:27]: >'Interrupt signal' [](http://example.com/codeflow?start=71&length=18)

This is not a crash. It's a OSConfig termination by Ctrl +C when console app for example, or stop service.

File: .github/actions/check-logs/action.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        crash_found = False
+
+        for log_file in log_files:
+            with open(log_file, 'r') as f:
+                for line_num, line in enumerate(f, 1):
+                    for crash in crashes:
+                        if crash in line:
+                            error(crash, os.path.basename(log_file), line_num, line)
```
Comment (MariusNi) [inline .github/actions/check-logs/action.yml:35]: >error(crash, os.path.basename(log_file), line_num, line) [](http://example.com/codeflow?start=28&length=56)

The line with the crash in the agent and the platform logs is not normally logged. It is written raw before the process terminates. Thus we should not expect any time/date/file/line stamps.

**Review by MariusNi — APPROVED** [review]
:shipit:

File: .github/actions/check-logs/osconfig_platform.log [hunk 3]
```diff
@@ -0,0 +1,81 @@
+[2022-06-08 22:21:15] [MpiServer.c:95] MpiGet(DeviceInfo, osConfigVersion) request, session 0x7fb71c012e80 ('D3E0EDAA-3AA4-BB66-4B96-7022165EA3F8')
```
[the remark this answers is not in the document]
Comment (robertschaedler3) [reply .github/actions/check-logs/osconfig_platform.log:1]: Yes, just using for testing for now to see how everything will render 😄, thanks!

**Pull request merged**